### PR TITLE
Add add-client feature with car support

### PR DESCRIPTION
**Pull request by limyuquan**
Fixes: #44 
Important logic to review:

Duplicate cars are not allowed to be created during add-client:
 - Duplicate cars are: Cars with same VIN OR VRN
 - If there are other people with these duplicate VIN OR VRN, Exception raised

- Didnt delete tags due to future possibility of tags being used for issues

- After merging and pulling, you should delete your local addressBook.json to populate with the new fields, but not doing so is also fine.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AddClientCommand.java [hunk 1]
```diff
@@ -14,27 +17,34 @@
...
+            + PREFIX_MAKE + " VEHICLE MAKE (OPTIONAL) "
+            + PREFIX_MODEL + " VEHICLE MODEL (OPTIONAL) \n"
             + "Example: " + COMMAND_WORD + " "
             + PREFIX_NAME + "John Doe "
             + PREFIX_PHONE + "98765432 "
             + PREFIX_EMAIL + "johnd@example.com "
             + PREFIX_ADDRESS + "311, Clementi Ave 2, #02-25 "
-            + PREFIX_TAG + "friends "
-            + PREFIX_TAG + "owesMoney";
+            + PREFIX_VRN + "SGX1234A "
```
Comment (xhamyo) [inline src/main/java/seedu/address/logic/commands/AddClientCommand.java:41]: I think the VRN example should be SGX 1234 B (with space and a valid VRN).

File: src/main/java/seedu/address/logic/commands/AddClientCommand.java [hunk 2]
```diff
@@ -14,27 +17,34 @@
...
+            + PREFIX_VIN + "12345678901234567 "
```
Comment (xhamyo) [inline src/main/java/seedu/address/logic/commands/AddClientCommand.java:42]: I think the VIN example can be a real one KMHGH4JH3EU073801 for clarity.

File: src/main/java/seedu/address/logic/commands/AddClientCommand.java [hunk 3]
```diff
@@ -14,27 +17,34 @@
...
+            + PREFIX_MAKE + "Toyota "
+            + PREFIX_MODEL + "Corolla ";
 
+    public static final String VEHICLE_DETAILS_MISSING =
+        "Vehicle details are missing. Please provide all vehicle details. \n" + MESSAGE_USAGE;
```
Comment (xhamyo) [inline src/main/java/seedu/address/logic/commands/AddClientCommand.java:47]: I think wrapped line should have 2 tabs.

File: src/test/java/seedu/address/model/car/CarTest.java [hunk 4]
```diff
@@ -9,24 +9,30 @@
 public class CarTest {
 
     private static final Car ALICE_HYUNDAI_KONA = new Car(new Vrn("SJH 9514 P"),
-            new Vin("KMHGH4JH3EU073801"),
-            new CarMake("Hyundai"),
-            new CarModel("Kona 1.6T"));
+        new Vin("KMHGH4JH3EU073801"),
```
Comment (xhamyo) [inline src/test/java/seedu/address/model/car/CarTest.java:12]: I think wrapped line should have 2 tabs.

File: src/test/java/seedu/address/testutil/TypicalPersons.java [hunk 5]
```diff
@@ -24,36 +32,54 @@
 public class TypicalPersons {
 
     public static final Person ALICE = new PersonBuilder().withName("Alice Pauline")
-            .withAddress("123, Jurong West Ave 6, #08-111").withEmail("alice@example.com")
-            .withPhone("94351253")
-            .withTags("friends").build();
+        .withAddress("123, Jurong West Ave 6, #08-111").withEmail("alice@example.com")
```
Comment (xhamyo) [inline src/test/java/seedu/address/testutil/TypicalPersons.java:35]: I think wrapped line should have 2 tabs.

File: src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java [hunk 6]
```diff
@@ -38,62 +53,149 @@ public void toModelType_validPersonDetails_returnsPerson() throws Exception {
...
+        assertEquals(BENSON_WITH_CAR, person.toModelType());
+    }
+
+    /*
+     * Test for missing Car Fields
+     */
+    @Test
+    public void toModelType_missingVin_throwsIllegalValueException() {
+        JsonAdaptedPerson person = new JsonAdaptedPerson(VALID_NAME, VALID_PHONE, VALID_EMAIL, VALID_ADDRESS,
+                                            VALID_TAGS, null, VALID_CAR_VRN_A, VALID_CAR_MAKE_A, VALID_CAR_MODEL_A);
```
Comment (xhamyo) [inline src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java:68]: Not 100% sure, but might need to make this 2 tabs instead.

**Review by xhamyo — APPROVED** [review]
Logic LGTM, will good to update example and code style.

**Review by calebyyy — APPROVED** [review]
LGTM

**Pull request merged**